pyfr/plugins/fluidforce.py

```diff
@@ -117,10 +139,18 @@ def __call__(self, intg):
 
         # Solution matrices indexed by element type
         solns = dict(zip(intg.system.ele_types, intg.soln))
-        ndims, nvars = self.ndims, self.nvars
+        ndims, nvars, mcomp = self.ndims, self.nvars, self._mcomp
+
+        # Force and moment vectors
+        n = 2 if self._viscous else 1
+        fm = np.zeros((ndims + mcomp, n))
 
-        # Force vector
-        f = np.zeros(2*ndims if self._viscous else ndims)
+        # Pressure and viscous components
+        fp, mp = fm[:ndims, 0], fm[ndims:, 0]
+        if self._viscous:
```

Comment:
Can we avoid defining fv and mv?

Reply:
As it was previously with just `f` and `m` for both viscous and pressure forces/moments?

Reply:
Accumulating into fm[...] directly.